Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableMediator.cpp

```diff
@@ -0,0 +1,227 @@
+                ->Method("CreateSpawnTicket", &SpawnableMediator::CreateSpawnTicket)
+                ->Method("Spawn", &SpawnableMediator::Spawn)
+                ->Method("SpawnAndParent", &SpawnableMediator::SpawnAndParent)
+                ->Method("SpawnParentAndTransform", &SpawnableMediator::SpawnParentAndTransform)
+                ->Method("Despawn", &SpawnableMediator::Despawn)
+                ->Method("Clear", &SpawnableMediator::Clear);
+
+            behaviorContext->EBus<SpawnableNotificationsBus>("SpawnableNotificationsBus")
+                ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Common)
+                ->Attribute(AZ::Script::Attributes::Category, "Prefab/Spawning")
+                ->Attribute(AZ::Script::Attributes::Module, "prefabs")
+                ->Handler<SpawnableNotificationsHandler>();
+        }
+    }
+
+    SpawnableMediator::SpawnableMediator()
+        : m_interface(SpawnableEntitiesInterface::Get())
+    {
+    }
+
+    SpawnableMediator::~SpawnableMediator()
+    {
+        Clear();
+    }
+
+    void SpawnableMediator::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
+    {
+        ProcessDespawnedResults();
+        ProcessSpawnedResults();
+        AZ::TickBus::Handler::BusDisconnect();
+    }
+
+    EntitySpawnTicket SpawnableMediator::CreateSpawnTicket(const SpawnableAssetRef& spawnableAsset)
+    {
+        return EntitySpawnTicket(spawnableAsset.m_asset);
+    }
+
+    bool SpawnableMediator::Spawn(EntitySpawnTicket spawnTicket)
+    {
+        static AZ::EntityId parentId;
+        return SpawnParentAndTransform(
+            spawnTicket,
+            parentId,
+            AZ::Vector3::CreateZero(), // translation
+            AZ::Vector3::CreateZero(), // rotation
+            1.0f); // scale
+    }
+
+    bool SpawnableMediator::SpawnAndParent(EntitySpawnTicket spawnTicket, const AZ::EntityId& parentId)
+    {
+        return SpawnParentAndTransform(
+            spawnTicket,
+            parentId,
+            AZ::Vector3::CreateZero(), // translation
+            AZ::Vector3::CreateZero(), // rotation
+            1.0f); // scale
+    }
+
+    bool SpawnableMediator::SpawnParentAndTransform(
+        EntitySpawnTicket spawnTicket,
+        const AZ::EntityId& parentId,
+        const AZ::Vector3& translation,
+        const AZ::Vector3& rotation,
+        float scale)
+    {
+        if (!spawnTicket.IsValid())
+        {
+            AZ_Error("Spawnables", false, "EntitySpawnTicket used for spawning is invalid.");
+            return false;
+        }
+
+        auto preSpawnCB = [parentId, translation, rotation, scale](
+                              [[maybe_unused]] EntitySpawnTicket::Id ticketId,
+            SpawnableEntityContainerView view)
+        {
+            AZ::Entity* rootEntity = *view.begin();
+            TransformComponent* entityTransform = rootEntity->FindComponent<TransformComponent>();
```

Comment:
This will only parent the first entity in the spawnable to the provided parent. What about the rest of the entities? Can you explain the logic here and also add a comment accordingly?

Reply:
it will parent the container entity, I believe this should work

Reply:
I will just rename rootEntity to containerEntity, I think this would be clearer.